datumaro/components/annotation.py

```diff
@@ -266,46 +328,72 @@ def from_instance_masks(instance_masks,
         else:
             instance_labels = [None] * len(instance_masks)
 
-        instance_masks = sorted(enumerate(instance_masks),
-            key=lambda m: m[1].z_order)
-        instance_masks = ((m.image, 1 + j,
-                instance_ids[i] if instance_ids[i] is not None else 1 + j,
-                instance_labels[i] if instance_labels[i] is not None else m.label
-            ) for j, (i, m) in enumerate(instance_masks))
+        max_index = len(instance_masks) + 1
```

Comment:
You can't apply `len` to an `Iterable`. You have to either convert it to a list, or change the type to something more restrictive (e.g. `Sequence`).

Reply:
Fixed